### PR TITLE
Add data terminals and PNET packet capabilities to cloning console

### PR DESCRIPTION
**Pull request by glassofmilk**
[ENHANCEMENT]

## About the PR 
This adds wired packet capabilities to the cloning console as well as placing a data terminal under the cloning consoles on each map so they can take advantage of this new feature. New cloner setups can be created WITHOUT requiring a data terminal. I wanted to make sure this wouldn't make setting up an auxiliary cloner more difficult!

The cloning console uses standard term_connect protocol, so you can connect to it with a DWAINE terminal. This makes using packets to communicate with it a little trickier, which is probably good since you could theoretically use this to build a resurrection machine. It allows you to give the following commands to the cloning console: list_records, details (passing in the record id or name), and clone (passing in the record id or name).

## Why's this needed? 
I think we need more good payoffs for playing with packets that don't actively ruin other players' days. Also, I think the idea of an traitor setting up a secret cloner with a MechComp contraption to clone them every time they die a glorious death is really funny.

## Changelog

```
(u)glassofmilk:
(*)Added wired packet capabilities to the cloning console. Create your own Cylon-style resurrection machine!
```

### CODE REVIEW

**Comment by flrsh** [comment]
I'm not a fan of this change. Cloning consoles don't thematically fit as a networked device, and I don't think you should be able to clone yourself with packets.

**Comment by keiyakins** [comment]
IMO they'd fit perfectly - it'd make sense to back up clonescans to the computer core, and you should really be able to pull up medical records from them, from a UX perspective. Whether they fit gamewise is another question I'm not really qualified to answer.

**Comment by glassofmilk** [comment]
Even with this patch, the cloner doesn't do much with the mainframe. I'd like to expand on this idea more and move some of the communication between cloner components into packets as well as storing data on the mainframe, but I haven't started on that yet.

File: code/obj/machinery/computer/cloning.dm [hunk 1]
```diff
@@ -78,9 +82,152 @@
...
+
+		if(!src.link)
+			var/turf/T = get_turf(src)
+			var/obj/machinery/power/data_terminal/test_link = locate() in T
+			if(test_link && !DATA_TERMINAL_IS_VALID_MASTER(test_link, test_link.master))
+				src.link = test_link
+				src.link.master = src
+
+		src.net_id = generate_net_id(src)
 		return
```
Comment (ZeWaka) [inline code/obj/machinery/computer/cloning.dm:94]: Unnecessary.

File: code/obj/machinery/computer/cloning.dm [hunk 2]
```diff
@@ -78,9 +82,152 @@
...
 	return
```
Comment (ZeWaka) [inline code/obj/machinery/computer/cloning.dm:95]: Unnecessary.

File: code/obj/machinery/computer/cloning.dm [hunk 3]
```diff
@@ -78,9 +82,152 @@
...
+
+					if(!foundRecord)
+						post_status(target, "command", "term_message", "data", "message=No record found")
+						return
+
+					var/obj/item/implant/health/H = locate(foundRecord.fields["imp"])
+					var/healthStatus = ((H) && (istype(H))) ? H.sensehealth() : "Unable to locate implant"
+					post_status(target, "command", "term_message", "data", "id=[foundRecord.fields["id"]]&name=[foundRecord.fields["name"]]&health=[healthStatus]")
+					return
+	return
```
Comment (ZeWaka) [inline code/obj/machinery/computer/cloning.dm:205]: Unnecessary.

**Comment by flrsh** [comment]
Thematically, the station network and its devices are research oriented and archaic. The cloning console and its accompanying machines feel a lot more modern. 

The mainframe facilitates communication between devices on the network. Being able to directly connect to a cloning console without that intermediary feels really weird. The "you can set this up with a data terminal but you don't have to" state also feels weird.

This functionality is neat, but I think there are more fitting ways of making packets more interesting. Maybe change this from wired to wireless (to me there's a large difference in terms of how they feel), and focus more on the cloning scanner and pod than the console itself? You could have signals for when people are scanned, start cloning, when the biomatter has depleted, when they pop out, etc.

**Comment by ZeWaka** [comment]
>The "you can set this up with a data terminal but you don't have to" state also feels weird.
> 
> You could have signals for when people are scanned, start cloning, when the biomatter has depleted, when they pop out, etc.

+1 to these

**Pull request closed without merging**